packages/xo-web/src/xo-app/sr/tab-disks.js

```diff
@@ -319,6 +409,89 @@ export default class SrDisks extends Component {
     vbdsByVdi => mapValues(vbdsByVdi, vbds => some(vbds, 'attached'))
   )
 
+  _getVmSrs = createSelector(
+    _ => _,
+    () => this.props.vbds,
+    () => this.props.allVmsByVdi,
+    () => this.props.allVdis,
+    () => this.props.srs,
+    (selectedVdis, allVbds, vmsByVdi, vdis, srs) => {
+      let vbds = []
+      forEach(selectedVdis, ({ id }) => {
+        forEach(vmsByVdi[id], vm => {
+          vbds = [...vbds, ...vm.$VBDs]
+        })
+      })
+
+      return compact(
+        vbds.map(vbdId => {
+          const vbd = allVbds[vbdId]
+          let vdi
+          return (
+            !vbd.is_cd_drive &&
+            ((vdi = vdis[vbd.VDI]), vdi !== undefined && srs[vdi.$SR])
+          )
+        })
+      )
+    }
+  )
+
+  _getRequiredHost = createSelector(this._getVmSrs, srs => {
+    if (!areSrsOnSameHost(srs)) {
+      return
+    }
+
+    let container
+    forEach(srs, sr => {
+      if (sr !== undefined && !isSrShared(sr)) {
+        container = sr.$container
+        return false
+      }
+    })
+    return container
+  })
```

Comment:
I don't really get the process there. There isn't only one "required host" since there can be multiple VMs. And what if we want to migrate all the VDIs attached to a VM from a SR to another SR, this shouldn't trigger a warning, right? Could you describe, in the PR description, exactly what you're checking and which case triggers a warning?

Reply:
I take into account all your questions and I wrote a clearer code.

Reply:
If we want to migrate all the VDIs attached to a VM from a SR to another SR:
    - If the SR destination is shared: No warning.
    - If the selected VDIs are attached to other VMs? response in the PR description.
    - If these VDIs are attached to a unique VM:  it's a particuliar case and for now the warning will be displayed.